eclipse-collections/src/main/java/org/eclipse/collections/impl/lazy/FlatCollectIterable.java

```diff
@@ -78,6 +79,24 @@ public V detect(Predicate<? super V> predicate)
     }
 
     @Override
+    public Optional<V> detectOptional(Predicate<? super V> predicate)
+    {
+        V[] result = (V[]) new Object[1];
+        Iterate.detectOptional(this.adapted, each -> {
```

Comment:
This call is actually logically significant in a way that it should throw NullPointerException when it detects the value which is null. Throwing NullPointerException is handled within Iterate.detectOptional(), when Optional.of() is called.

Having said that, we could change the detect() side as you suggest, but maybe keeping as is makes the structure consistent. Let me know.